docs/specs/multi-registry/multi_registry.md

```diff
@@ -0,0 +1,366 @@
+  be changed. Weaver will detect and report an error if an experimental entity is referenced across registries.
+
+#### Do we allow circular references between registries?
+- Short answer: No.
+- Long answer: Circular references are already prohibited inside a registry, so circular references between
+  registries can lead to the kind of problems (e.g. unresolvable references). Weaver will detect and report
+  an error if a circular reference is detected between registries.
+
+#### Do we allow references to groups or attributes defined in an imported registry?
+- Short answer: Yes.
+- Long answer: References to groups or attributes defined in an imported registry are allowed. This feature
+  allows a registry to extend or override the attributes of an imported group. Weaver will detect and report an
+  error if a reference is made to an entity that is not defined in the imported registry.
+
+### Open Questions:
+- Can we override any field of a group defined in an imported registry? No for `type` field, what about the other
+  fields?
+- Is there a relationship to define between the instrumentation scope name and version and the registry?
+
+
+## Core Policies
+
+These core policies aim to ensure consistency and backward compatibility across registries.
+
+### Group Policies
+- No group ID collisions
+- Only attribute groups are allowed in the attribute registry
+- Ref attributes are not allowed in the attribute registry
+- ID attributes are not allowed in groups outside of the attribute registry
+
+### Attribute Policies
+- No attribute name collisions
+- Attributes cannot be removed
+- Attributes cannot "degrade" in stability (e.g., stable -> experimental)
+- Stable attributes cannot change type
+- All attributes must be defined inside a semantic convention registry
+- Attribute names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+- Attributes with prefixes are not allowed; attributes must be fully qualified
+
+### Enum Member Policies
+- Stable members cannot change stability
+- Values cannot change
+- IDs cannot be removed
+- Enum IDs must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Metrics
+- No metric name collisions
+- Metrics cannot be removed
+- Metrics cannot "degrade" in stability (e.g., stable -> experimental)
+- Stable metric units cannot change
+- Stable metric instruments cannot change
+- The set of required/recommended attributes must remain the same
+- Metric names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Events
+- No event name collisions
+- Events cannot be removed
+- Events cannot "degrade" in stability (e.g., stable -> experimental)
+- The set of required/recommended attributes must remain the same
+- Event names must match the following regular expression: `^[a-z][a-z0-9]*([._][a-z0-9]+)*$`
+
+### Spans
+- No span name collisions
+- Spans cannot be removed
```

Comment:
why not? what does it even mean to remove a span?

I assume I pick the spans that I want to support by referencing them and I should be free to drop any spans that I don't need.

Moreover, if I'm defining semconv for a specific system (e.g. Kafka library), not all spans declared in generic conventions apply to my library, some of them are defined for scenarios that I don't support and it'd be best if I don't ever mention corresponding spans in my conventions.

Reply:
In the OTEL registry, once a span has been defined, can its definition be removed in a later version? I thought this wasn't allowed if the span had a stability status set to `stable`. If that's the case, I think we should apply the same rule for all registries, whether OTEL or not.

Reply:
Let's say I work on a library that supports multiple protocols (HTTP and something else). At some point my library removes support for one of these protocols. I should be allowed to stop referencing corresponding protocol spans - I no longer report them.

For metrics, for example, we at least support informal requirement levels and we could say that if you report e.g. http metrics, you must at least report at least required ones. Even this is tough to say - how do you know that they are going to report HTTP metrics if they don't reference them?

For spans it's even more complicated since span identity is not defined and spans don't have requirement levels. 
We don't define anything about spans in yaml beyond what's defined for the attribute_groups - we need to change this first.

----------

I think there are two separate discussions here:
1. What's allowed when I reference conventions from registry A in registry B (or within one registry). 
2. What changes are allowed in the the registry B from version N to N+1

I'm mostly talking about p1 (you can reference or not reference any group), and I think you're talking about p2. 

Maybe it'd be more clear if policies were separated into two groups: within-one-version and backward-compatibility?